interop-cats/shared/src/main/scala/zio/interop/cats.scala

```diff
@@ -385,3 +388,39 @@ final private class CatsContravariant[E, T] extends Contravariant[ZIO[*, E, T]]
   override def contramap[A, B](fa: ZIO[A, E, T])(f: B => A): ZIO[B, E, T] =
     ZIO.accessM[B](b => fa.provide(f(b)))
 }
+
+final private class NonEmptyChunkReducible extends Reducible[NonEmptyChunk] {
+
+  override def foldLeft[A, B](fa: NonEmptyChunk[A], b: B)(f: (B, A) => B): B = fa.foldLeft(b)(f)
+  override def foldRight[A, B](fa: NonEmptyChunk[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] = {
+    val idx = fa.length - 1
+    val acc = lb
+    def loop(idx: Int, acc: Eval[B]): Eval[B] =
+      if (idx < 0) acc
+      else loop(idx - 1, Eval.defer(f(fa(idx), acc)))
+
+    Eval.defer(loop(idx, acc))
+  }
+
+  override def reduceLeftTo[A, B](fa: NonEmptyChunk[A])(f: A => B)(g: (B, A) => B): B =
+    fa.tail.foldLeft(f(fa.head))(g)
+
+  override def reduceRightTo[A, B](fa: NonEmptyChunk[A])(f: A => B)(g: (A, Eval[B]) => Eval[B]): Eval[B] = {
+    val idx = fa.length - 2
+    val acc = f(fa.last)
+    def loop(idx: Int, acc: Eval[B]): Eval[B] =
+      if (idx < 0) acc
+      else loop(idx - 1, Eval.defer(g(fa(idx), acc)))
+
+    Eval.defer(loop(idx, Eval.now(acc)))
+  }
+
+}
+
+final private class NonEmptyChunkEq[A] extends Eq[NonEmptyChunk[A]] {
+
+  override def eqv(x: NonEmptyChunk[A], y: NonEmptyChunk[A]): Boolean =
```

Comment:
I think we want to require an `Eq[A]` here and use that to compare pairwise elements instead of universal equality.